src/app/client/src/app/modules/dashboard/components/filter/filter.component.ts

```diff
@@ -106,21 +107,50 @@ export class FilterComponent implements OnInit, OnDestroy {
     this.unsubscribe.complete();
   }
 
-  formGeneration(chartData) {
-    this.filtersFormGroup = this.fb.group({});
-    _.forEach(this.filters, filter => {
-      if (filter.controlType === 'date' || /date/i.test(_.get(filter, 'reference'))) {
-        const dateRange = _.uniq(_.map(chartData, _.get(filter, 'reference')));
-        this.pickerMinDate = moment(dateRange[0], 'DD-MM-YYYY');
-        this.pickerMaxDate = moment(dateRange[dateRange.length - 1], 'DD-MM-YYYY');
-        this.dateFilterReferenceName = filter.reference;
+  formGeneration(chartData, resetOptions = false) {
```

Comment:
Put this logic inside a separate method since a method should do only one job and this method is meant for form creation.

Put the form controls update logic inside a different method.

Reply:
resolved